mipengine/node/monetdb_interface/common.py

```diff
@@ -0,0 +1,240 @@
+
+import pymonetdb
+from pymonetdb.sql.cursors import Cursor
+
+from mipengine.common.node_catalog import NodeCatalog
+from mipengine.node.config.config_parser import Config
+from mipengine.node.node import config
+from mipengine.node.tasks.data_classes import ColumnInfo
+
+MONETDB_VARCHAR_SIZE = 50
+
+# TODO Add monetdb asyncio connection (aiopymonetdb)
+config = Config().config
+node_catalog = NodeCatalog()
+local_node = node_catalog.get_local_node_data(config["node"]["identifier"])
+monetdb_hostname = local_node.monetdbHostname
+monetdb_port = local_node.monetdbPort
+connection = pymonetdb.connect(username=config["monet_db"]["username"],
+                               port=monetdb_port,
+                               password=config["monet_db"]["password"],
+                               hostname=monetdb_hostname,
+                               database=config["monet_db"]["database"])
+cursor: Cursor = connection.cursor()
+
+
+def create_table_name(table_type: str, context_id: str, node_id: str) -> str:
+    """
+    Creates a table name with the format <table_type>_<context_id>_<node_id>_<uuid>
+    """
+    if table_type not in {"table", "view", "merge"}:
+        raise KeyError(f"Table type is not acceptable: {table_type} .")
+    if node_id not in {"global", config["node"]["identifier"]}:
+        raise KeyError(f"Node Identifier is not acceptable: {node_id} .")
+
+    uuid = str(pymonetdb.uuid.uuid1()).replace("-", "")
+
+    return f"{table_type}_{context_id}_{node_id}_{uuid}"
+
+
+# TODO Add SQLAlchemy if possible
+# TODO We need to add the PRIVATE/OPEN table logic
+
+def get_table_type_enumeration_value(table_type: str) -> int:
+    """ Converts MIP Engine's table types to MonetDB's table types
+     normal -> 0,
+     view -> 1,
+     merge -> 3,
+     remote -> 5,
+        """
+    return {
+        "normal": 0,
+        "view": 1,
+        "merge": 3,
+        "remote": 5,
+    }[table_type]
+
+
+def convert_to_monetdb_column_type(column_type: str) -> str:
+    """ Converts MIP Engine's int,float,text types to monetdb
+    int -> integer
+    float -> double
+    text -> varchar(50s)
+    bool -> boolean
+    clob -> clob
+    """
+    return {
+        "int": "int",
+        "float": "double",
+        "text": f"varchar({MONETDB_VARCHAR_SIZE})",
+        "bool": "bool",
+        "clob": "clob",
+    }[str.lower(column_type)]
+
+
+def convert_from_monetdb_column_type(column_type: str) -> str:
+    """ Converts MonetDB's types to MIP Engine's types
+    int ->  int
+    double  -> float
+    varchar(50)  -> text
+    boolean -> bool
+    clob -> clob
+    """
+    return {
+        "int": "int",
+        "double": "float",
+        "varchar": "text",
+        "bool": "bool",
+        "clob": "clob",
+    }[str.lower(column_type)]
+
+    if column_type not in type_mapping.keys():
+        raise TypeError(f"Type {column_type} cannot be converted to monetdb column type.")
+
+    return type_mapping.get(column_type)
```

Comment:
To have one method that holds the transformation logic, that could be used from multiple places. If in the future we change float to real for example, we will only have to adjust one place.

Reply:
Type conversions between mip-engine and monetdb will always be a dict lookup, no? If you change float to real you just change the dict entry. I am not sure I understand your point.

Reply:
Where would that dict exist?

You are just suggesting to move it out of the method?

Reply:
Yes, you move it out, give it a name a use it instead of the function. These functions are just wrappers for dict lookups. In my opinion they are redundant.

Reply:
These functions will not only be the lookup. It will validate that the key exists as well and throw an error if not. Otherwise you would have to do it when you are using it every time.

`def convert_from_monetdb_column_type(column_type: str) -> str:
    """ Converts MonetDB's types to MIP Engine's types
    int ->  int
    double  -> float
    varchar(50)  -> text
    boolean -> bool
    clob -> clob
    """
    type_mapping = {
        "int": "int",
        "double": "float",
        "varchar": "text",
        "bool": "bool",
        "clob": "clob",
    }

    if column_type not in type_mapping.keys():
        raise TypeError(f"Type {column_type} cannot be converted to monetdb column type.")

    return type_mapping.get(column_type)`

Personally, I prefer it like that.

Reply:
A dict raises a KeyError if the key doesn't exist anyway. You are just emulating a dict's behavior, no new functionality whatsoever. 
It's like having this function
```
def none():
    return None
```
and using it instead of `None`.

Reply:
Not exactly, instead of writing:

```
    if column_type not in monetdb_column_types.keys():
        raise TypeError(f"Type {column_type} cannot be converted to monetdb column type.")
    type = type_mapping.get(column_type)
```

on multiple places in the code, we will only write:

```
    type = convert_from_monetdb_column_type(monet_db_type)
```

1) We won't need to check if the key exists every time,
2) we won't have a different error message every time and
3) if we want to change the dict into something else for some reason, or add a different behavior, it will only happen in one place.